File: test-case/char/chars1d/chars1d.php
<?php
require_once 'Soda/Unittest/index.php';

use Soda\Unittest\TestWork;
use Soda\Unittest\Utils;

class Solution {

    /**
     * @param String[] $chars
     * @return String[]
     */
    function doubleList($chars) {
        $arr = array();
        array_push($arr, ...$chars);
        array_push($arr, ...$chars);
        return $arr;
    }
}

$work = TestWork::create(new Solution(), 'doubleList', $argv[0]);
// $work->validator = fn($e, $r) => <boolean result>
$work->compareSerial = true;
echo $work->run(Utils::fromStdin());

File: test-case/char/chars1d/Chars1d.kt
import soda.kotlin.unittest.*

class Solution {
    fun doubleList(chars: CharArray): List<Char> {
        val res = mutableListOf<Char>()
        for (ch in chars) {
            res.add(ch)
        }
        for (ch in chars) {
            res.add(ch)
        }
        return res
    }
}

class Chars1d : (String) -> String {
    override fun invoke(text: String): String {
        val work = GenericTestWork.create(Solution()::doubleList)

        // * by method that has not return value
        // val work = GenericTestWork.createVoid(Solution.METHOD_WITHOUT_RETURN)
        // * by class of data struct
        // val work = GenericTestWork.forStruct(STRUCT::class)

        // * setup validator
        // work.validator = (T, T) -> Boolean
        work.compareSerial = true
        return work.run(text)
    }

}

fun main() {
    print(Chars1d()(Utils.fromStdin()))
}
